quinn-proto/src/endpoint.rs

```diff
@@ -321,25 +321,21 @@ impl Endpoint {
     /// Initiate a connection
     pub fn connect(
         &mut self,
-        log: Option<Logger>,
+        config: ClientConfig,
```

Comment:
I'm not totally sure how I feel about taking this by value when we take `ServerConfig` through an `Arc`, but doing that here feels silly because all we do is pull out the fields; it'd be strictly more work.